onnxruntime/core/providers/cuda/tensor/scatter_elements_impl.h

```diff
@@ -24,6 +24,18 @@ Status ScatterElementsImpl(
     const int axis,
     T* output_data);
 
+template <typename T, typename Tin>
+Status GatherElementsGradImpl(
+    const int rank,
+    TArray<int64_t>& buffer_input_dims,
+    TArray<int64_t>& buffer_input_strides,
+    const Tin* indices_data,
+    const int64_t indices_size,
+    TArray<int64_t>& buffer_indices_dims,
+    TArray<fast_divmod>& indices_strides,
+    const T* updates,
+    const int axis,
+    T* output_data);
```

Comment:
same for this. 
Avoid building this in the inference only build

Reply:
Good point! Yeah we can use RNABLR_TRAINING to control this.